### PR TITLE
[DEV-3028] link and payment external id

### PR DESCRIPTION
**Pull request by bernd2022**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Review by TaprootFreak — APPROVED** [review]

**Comment by bernd2022** [comment]
I don't think it makes sense to allow the “PaymentExternalId” in the “createPayment” request.

**Comment by TaprootFreak** [comment]
Ich find schon 😄

**Comment by TaprootFreak** [comment]
ok, ich glaub jetzt verstehe ichs

File: src/subdomains/core/payment-link/controllers/payment-link.controller.ts [hunk 1]
```diff
@@ -36,16 +36,20 @@ export class PaymentLinkController {
...
   @ApiQuery({ name: 'id', description: 'Link ID', required: false })
-  @ApiQuery({ name: 'externalId', description: 'External link ID', required: false })
+  @ApiQuery({ name: 'externalLinkId', description: 'External link ID', required: false })
+  @ApiQuery({ name: 'externalPaymentId', description: 'External payment ID', required: false })
   async getAllPaymentLinks(
     @GetJwt() jwt: JwtPayload,
     @Query('id') id: string,
-    @Query('externalId') externalId: string,
+    @Query('externalLinkId') externalLinkId: string,
+    @Query('externalPaymentId') externalPaymentId: string,
```
Comment (davidleomay) [inline src/subdomains/core/payment-link/controllers/payment-link.controller.ts:45]: Rename `id` to `linkId` everywhere

File: src/subdomains/core/payment-link/services/payment-link-payment.service.ts [hunk 2]
```diff
@@ -96,6 +96,15 @@ export class PaymentLinkPaymentService {
     });
   }
 
+  async getPaymentByExternalId(externalPaymentId: string): Promise<PaymentLinkPayment | null> {
+    return this.paymentLinkPaymentRepo.findOne({
+      where: { externalId: externalPaymentId },
+      relations: {
+        link: true,
+      },
```
Comment (davidleomay) [inline src/subdomains/core/payment-link/services/payment-link-payment.service.ts:104]: This relation shouldn't be necessary?

**Review by davidleomay — APPROVED** [review]

**Pull request merged**